Engine/EditorUI/Panel/InspectorSubpanel/PanelComponent.cpp

```diff
@@ -332,57 +316,62 @@ void PanelComponent::ShowComponentBillboard(ComponentBillboard* billboard)
-		if (selected_resource_uuid != 0)
-		{
-			billboard->ChangeTexture(selected_resource_uuid);
-		}
-		selected_resource_uuid = ImGui::ResourceDropper<Texture>();
-		if (selected_resource_uuid != 0)
-		{
-			billboard->ChangeTexture(selected_resource_uuid);
-		}
-		int alignment_type = static_cast<int>(billboard->alignment_type);
-		if (ImGui::Combo("Billboard type", &alignment_type, "View point\0Axial\0Spritesheet\0Not aligned")) {
-			switch (alignment_type)
-			{
-			case 0:
-				billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::VIEW_POINT);
-				break;
-			case 1:
-				billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::AXIAL);
-				break;
-			case 2:
-				billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::SPRITESHEET);
-				break;
-			case 3:
-				billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::CROSSED);
-				break;
-			}
-		}
 
-		if (billboard->alignment_type == ComponentBillboard::AlignmentType::SPRITESHEET) {
-			ImGui::InputInt("Rows", &billboard->x_tiles, 1);
-			ImGui::InputInt("Columns", &billboard->y_tiles, 1);
-			ImGui::InputFloat("Speed", &billboard->sheet_speed, 1);
-			ImGui::Checkbox("Oriented to camera", &billboard->oriented_to_camera);
+	ImGui::Text("Texture");
+	ImGui::SameLine();
+
+	std::string texture_name = billboard->billboard_texture == nullptr ? "None (Texture)" : App->resources->resource_DB->GetEntry(billboard->billboard_texture->GetUUID())->resource_name;
+	ImGuiID element_id = ImGui::GetID((std::to_string(billboard->UUID) + "TextureSelector").c_str());
+	if (ImGui::Button(texture_name.c_str()))
+	{
+		App->editor->popups->resource_selector_popup.ShowPanel(element_id, ResourceType::TEXTURE);
+	}
+
+	uint32_t selected_resource_uuid = App->editor->popups->resource_selector_popup.GetSelectedResource(element_id);
+	if (selected_resource_uuid != 0)
+	{
+		billboard->ChangeTexture(selected_resource_uuid);
+	}
+	selected_resource_uuid = ImGui::ResourceDropper<Texture>();
+	if (selected_resource_uuid != 0)
+	{
+		billboard->ChangeTexture(selected_resource_uuid);
+	}
+	int alignment_type = static_cast<int>(billboard->alignment_type);
+	if (ImGui::Combo("Billboard type", &alignment_type, "View point\0Axial\0Spritesheet\0Not aligned")) {
+		switch (alignment_type)
+		{
+		case 0:
+			billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::VIEW_POINT);
+			break;
+		case 1:
+			billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::AXIAL);
+			break;
+		case 2:
+			billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::SPRITESHEET);
+			break;
+		case 3:
+			billboard->ChangeBillboardType(ComponentBillboard::AlignmentType::CROSSED);
+			break;
 		}
-		
+	}
 
+	if (billboard->alignment_type == ComponentBillboard::AlignmentType::SPRITESHEET) {
+		ImGui::InputInt("Columns", &billboard->x_tiles, 1);
+		ImGui::InputInt("Rows", &billboard->y_tiles, 1);
```

Comment:
can you change this like i told you? rows for cols and cols for rows please :)

Reply:
I already change it, do you want me to change it back?